src/Detector/Ruby/RubyDetector.cs

```diff
@@ -0,0 +1,177 @@
+// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.Common.Extensions;
+using System;
+using System.Linq;
+
+namespace Microsoft.Oryx.Detector.Ruby
+{
+    /// <summary>
+    /// An implementation of <see cref="IPlatformDetector"/> which detects Ruby applications.
+    /// </summary>
+    public class RubyDetector : IRubyPlatformDetector
+    {
+        private readonly ILogger<RubyDetector> _logger;
+        private readonly DetectorOptions _options;
+
+        /// <summary>
+        /// Creates an instance of <see cref="RubyDetector"/>.
+        /// </summary>
+        /// <param name="logger">The <see cref="ILogger{RubyDetector}"/>.</param>
+        /// <param name="options">The <see cref="DetectorOptions"/>.</param>
+        public RubyDetector(ILogger<RubyDetector> logger, IOptions<DetectorOptions> options)
+        {
+            _logger = logger;
+            _options = options.Value;
+        }
+
+        public PlatformDetectorResult Detect(DetectorContext context)
+        {
+            bool isRubyApp = false;
+            string appDirectory = string.Empty;
+            var sourceRepo = context.SourceRepo;
+
+            if (sourceRepo.FileExists(RubyConstants.GemFileName))
+            {
+                isRubyApp = true;
+                _logger.LogInformation($"Found {RubyConstants.GemFileName} at the root of the repo.");
+            }
+            else
+            {
+                _logger.LogDebug(
+                    $"Could not find {RubyConstants.GemFileName} in repo");
+            }
+            
+            if (!isRubyApp) {
+                var isRubyLikeApp = false;
+                if (sourceRepo.FileExists(RubyConstants.GemFileLockName) 
+                    || sourceRepo.FileExists(RubyConstants.ConfigRubyFileName))
+                {
+                    isRubyLikeApp = true;
+                    _logger.LogInformation($"Found {RubyConstants.GemFileLockName}/{RubyConstants.ConfigRubyFileName} "
+                    + "at the root of the repo.");
+                }
+
+                if (isRubyLikeApp) {
```

Comment:
Yeah it's similar because presence of `gemfile ` can determine if it's a ruby app but presence of `gemfile.lock` and `config.ru` could not. So the logics quite same here.